src/components/operations/operation-details/ko/runtime/operation-details.ts

```diff
@@ -372,7 +376,7 @@ export class OperationDetails {
     }
 
     public async loadGatewayInfo(apiName: string): Promise<void> {
-        const hostnames = await this.apiService.getApiHostnames(apiName);
+        const hostnames = await this.apiService.getApiHostnames(apiName, this.includeAllHostnames());
```

Comment:
you are expanding the hostnames we are receiving in portal, but how do you make sure that we are using the correct one? In the following code, we are using the first one in the list.  So if  your new hostnames are in the end of the list, it wont be used. But in case that the list was empty, your hostname will be used. Is it the expected behaviour?

Reply:
This functionality was created with the goal of being able to select the default domain when multiple domains are declared. Currently if we declare multiple custom domains in Azure Portal, the default (Built-in) domain (<service-name>.azure-api.net) does not appear as an option in the Dev Portal.
![image](https://user-images.githubusercontent.com/88548172/217934167-b9f2c4ca-bf26-4f83-994d-80784125a203.png)

For this I had to update our endpoint in the backend to be able to receive this default domain in the Dev Portal.
https://msazure.visualstudio.com/DefaultCollection/One/_git/AAPT-APIManagement/pullrequest/7385078

Admin users can decide if they want this option enabled or not.
![image](https://user-images.githubusercontent.com/88548172/217934824-5ef6570e-a4c0-4e3b-9a03-6783943d89eb.png)

Users will be able to choose which domain to send requests to in the test console through this dropdown component.
![image](https://user-images.githubusercontent.com/88548172/217934891-83b9f165-30e4-4dd4-91ea-c2d468b41535.png)

I hope this can answer your question, let me know if more details are needed. Thank you!